native/src/main/java/io/ballerina/stdlib/persist/datastore/GoogleSheetsProcessor.java

```diff
@@ -0,0 +1,199 @@
+                                ModuleUtils.getModule(), Constants.PERSIST_GOOGLE_SHEETS_STREAM, sqlStream, targetType, 
+                                fields, includes, typeDescriptions, persistClient, null
+                        );
+
+                        RecordType streamConstraint =
+                                (RecordType) TypeUtils.getReferredType(targetType.getDescribingType());
+                        balFuture.complete(
+                                ValueCreator.createStreamValue(TypeCreator.createStreamType(streamConstraint,
+                                        PredefinedTypes.TYPE_NULL), persistStream)
+                        );
+                    }
+
+                    @Override
+                    public void notifyFailure(BError bError) {
+                        balFuture.complete(bError);
+                    }
+                }, null, streamTypeWithIdFields,
+                targetTypeWithIdFields, true, typeMap, true, fields, true, includes, true
+        );
+
+        return null;
+    }
+
+    public static Object queryOne(Environment env, BObject client, BArray path, BTypedesc targetType) {
+        BString entity = getEntity(env);
+        BObject persistClient = getPersistClient(client, entity);
+        BArray keyFields = (BArray) persistClient.get(KEY_FIELDS);
+        RecordType recordType = (RecordType) targetType.getDescribingType();
+
+        RecordType recordTypeWithIdFields = getRecordTypeWithKeyFields(keyFields, recordType);
+        BTypedesc targetTypeWithIdFields = ValueCreator.createTypedescValue(recordTypeWithIdFields);
+        ErrorType persistErrorType = TypeCreator.createErrorType(ERROR, ModuleUtils.getModule());
+        Type unionType = TypeCreator.createUnionType(recordTypeWithIdFields, persistErrorType);
+
+        BArray[] metadata = getMetadata(recordType);
+        BArray fields = metadata[0];
+        BArray includes = metadata[1];
+        BArray typeDescriptions = metadata[2];
+        BMap<BString, Object> typeMap = getFieldTypes(recordType);
+
+        Object key = getKey(env, path);
+
+        Future balFuture = env.markAsync();
+        env.getRuntime().invokeMethodAsyncSequentially(
+                getPersistClient(client, entity), Constants.RUN_READ_BY_KEY_QUERY_METHOD,
+                null, null, new Callback() {
+                    @Override
+                    public void notifySuccess(Object o) {
+                        balFuture.complete(o);
+                    }
+
+                    @Override
+                    public void notifyFailure(BError bError) {
+                        balFuture.complete(bError);
+                    }
+                },  null, unionType,
+                targetType, true, targetTypeWithIdFields, true, typeMap, true, key, true, fields, true, includes, true,
+                typeDescriptions, true
+        );
+
+        return null;
+    }
+}
+
+
```

Comment:
```suggestion
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)